### PR TITLE
Edit profile

### PR DESCRIPTION
**Pull request by lukaspode**
Captura de pantalla 2023-11-30 a la(s) 10 39 51
https://trello.com/c/S8s6i3Nc

### CODE REVIEW

File: app/views/users/_sidemenu.html.erb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      <div class="sidemenu-item">
+        <%= image_tag 'message_icon', class: 'sidemenu-icon'%>
+        <h2 class='sidemenu-title'>Messages</h2>
+      </div>
+
+      <div class="sidemenu-item">
+        <%= image_tag 'profile_icon', class: 'sidemenu-icon'%>
+        <h2 class='sidemenu-title'>Profile</h2>
+      </div>
+      <div class="sidemenu-item">
```
Comment (micaelalorenzo) [inline app/views/users/_sidemenu.html.erb]: Nit: Add empty line to keep consistency

File: app/views/users/_sidemenu.html.erb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  <div class="flex items-center">
+    <div class="rounded-full bg-slate-500 w-8 h-8 mr-3"></div>
+    <div class="mt-2 mb-3">
+      <h3 class="heading-5"><%= @user.name %></h3>
+      <p class="text-base font-medium text-dark-400">@<%= @user.username %></p>
+    </div>
+    <%= image_tag 'seemore_icon', class: 'ml-auto px-3'%>
+  </div>
+
+</div>
```
Comment (micaelalorenzo) [inline app/views/users/_sidemenu.html.erb]: I would remove this empty line.

File: app/javascript/application.js [hunk 3]
```diff
@@ -1,3 +1,9 @@
 // Configure your import map in config/importmap.rb. Read more: https://github.com/rails/importmap-rails
 import "@hotwired/turbo-rails"
 import "controllers"
+
+// document.addEventListener("turbo:frame-missing", (event) => {
+//   const { detail: { response, visit } } = event;
+//   event.preventDefault();
+//   visit(response);
+// });
```
Comment (micaelalorenzo) [inline app/javascript/application.js]: Is this commented code uploaded for some reason?

File: app/javascript/controllers/hello_controller.js [hunk 4]
```diff
@@ -1,7 +1,11 @@
...
-  connect() {
-    this.element.textContent = "Hello World!"
-  }
+  // connect() { }
+  // close(e){
+  //   e.preventDefault()
+  //   const modal = document.getElementById('modal')
+
+  //   modal.innerHTML = " "
+  // }
```
Comment (micaelalorenzo) [inline app/javascript/controllers/hello_controller.js]: Same here

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -1,4 +1,11 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::Base
+  private
+
+  def current_user?(user)
```
Comment (micaelalorenzo) [inline app/controllers/application_controller.rb]: I'm not sure if I would add this method here, where are you planning to use it?

File: app/javascript/application.js (hunk 3 above)
Comment (martin8767) [inline app/javascript/application.js]: do you need any of this ?

File: app/javascript/controllers/hello_controller.js (hunk 4 above)
Comment (martin8767) [inline app/javascript/controllers/hello_controller.js]: same

File: app/controllers/application_controller.rb [hunk 6]
```diff
@@ -1,4 +1,11 @@
...
+    user == current_user
+  end
+
+  helper_method :current_user?
```
Comment (martin8767) [inline app/controllers/application_controller.rb]:
```ruby
helper_method :current_user?

def current_user?(user)
  user == current_user
end
```

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -1,7 +1,41 @@
...
+
+  def user_params
+    params.require(:user).permit(:name, :bio, :website, :birth)
+  end
+
+  def user_equal_current_user?
+    return false if current_user?(@user)
+
+    redirect_to user_path(@user)
+  end
```
Comment (micaelalorenzo) [inline app/controllers/users_controller.rb]: What's the idea here? I think if you use the `current_user` in the update, you don't need to use this

File: app/controllers/users_controller.rb [hunk 8]
```diff
@@ -1,7 +1,41 @@
...
+
+  def set_user
     @user = User.find(params[:id])
   end
```
Comment (micaelalorenzo) [inline app/controllers/users_controller.rb]: We tend to use this kind of methods and then use it to access the user in controllers:
```ruby
def user
  @user ||= User.find(params[:id])
end
```
What `||=` does is assign the right "statement" to the variable only if the variable is falsely or nil. This way the first time you call the `user` method, `@user` will be nil and assign the result of `User.find(params[:id])` to it.

File: app/views/users/show.html.erb [hunk 9]
```diff
@@ -1,39 +1,33 @@
...
         <div class="flex justify-between w-full h-auto mb-1">
-          <div class="profile-picture">
-          </div>
-
-          <div class="mt-2.5">
-            <button class="btn btn-md btn-secondary">Edit Profile</button>
-          </div>
-
+          <%= image_tag 'profile_picture', class: 'profile-picture'%>
+          <%if current_user?(@user)%>
```
Comment (micaelalorenzo) [inline app/views/users/show.html.erb]: Add space before the `if`

File: app/views/users/show.html.erb [hunk 10]
```diff
@@ -1,39 +1,33 @@
...
+            <%= link_to "Edit Profile", edit_user_path(@user), class: 'btn btn-md btn-secondary sm:w-28 w-24 mt-2.5', data: { turbo_frame: 'modal' }%>
```
Comment (micaelalorenzo) [inline app/views/users/show.html.erb]: Add space after the `}`

File: app/views/users/show.html.erb [hunk 11]
```diff
@@ -1,39 +1,33 @@
...
+          <%end%>
```
Comment (micaelalorenzo) [inline app/views/users/show.html.erb]: And spaces between `end`

File: app/views/users/show.html.erb [hunk 12]
```diff
@@ -1,39 +1,33 @@
...
       <div class="mx-3.5">
         <div class="flex justify-between w-full h-auto mb-1">
-          <div class="profile-picture">
-          </div>
-
-          <div class="mt-2.5">
-            <button class="btn btn-md btn-secondary">Edit Profile</button>
-          </div>
-
+          <%= image_tag 'profile_picture', class: 'profile-picture'%>
```
Comment (micaelalorenzo) [inline app/views/users/show.html.erb]: Add space after `'profile-picture'`

File: config/routes.rb [hunk 13]
```diff
@@ -8,5 +8,10 @@
   }
   get 'up' => 'rails/health#show', as: :rails_health_check
 
-  resources :users, only: [:show]
+  devise_scope :user do
+    get 'signin' => 'users/sessions#create'
+    get 'signup' => 'users/registrations#new'
+  end
```
Comment (micaelalorenzo) [inline config/routes.rb]: I believe these routes are already initialized with `devise for :users` a couple of lines above. Do you need to re-defined them?

File: .DS_Store
Comment (micaelalorenzo) [inline .DS_Store:1]: You should add this types of file to the `.git_ignore`

File: app/controllers/users_controller.rb [hunk 14]
```diff
@@ -1,7 +1,41 @@
...
+  before_action :user_equal_current_user?, only: %i[edit update]
+
+  helper_method :current_user?
+
+  def show; end
+
+  def edit; end
+
+  def update
+    if @user.update(user_params)
```
Comment (micaelalorenzo) [inline app/controllers/users_controller.rb]: I would use `current_user` instead of @user

File: app/views/users/_sidemenu.html.erb [hunk 15]
```diff
@@ -0,0 +1,51 @@
+<div class="flex flex-col shrink-0 xl:w-[17.2rem] h-[56.25rem] w-[5.5rem] gap-[13.688rem] px-2.5">
+  <%# TOP %>
+  <div class="flex flex-col w-[15.938rem] h-[38.25rem]">
+    <div class="flex flex-row p-3 sm:block hidden">
+      <%= link_to image_tag('twitter_icon'), '#'%>
```
Comment (micaelalorenzo) [inline app/views/users/_sidemenu.html.erb]: Add space between `<%= something %>`

File: app/views/users/_sidemenu.html.erb [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+    </div>
+
+    <div class="flex flex-col items-start">
+      <div class="sidemenu-item">
+        <%= image_tag 'home_icon', class: 'sidemenu-icon'%>
```
Comment (micaelalorenzo) [inline app/views/users/_sidemenu.html.erb]: same here

File: app/views/users/_sidemenu.html.erb [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+        <h2 class='sidemenu-title'>Home</h2>
+      </div>
+
+      <div class="sidemenu-item">
+        <%= image_tag 'notification_icon', class: 'sidemenu-icon'%>
```
Comment (micaelalorenzo) [inline app/views/users/_sidemenu.html.erb]: Same here

File: app/views/users/_sidemenu.html.erb [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+        <p class='sidemenu-title'>Notifications</p>
+      </div>
+
+      <div class="sidemenu-item">
+        <%= image_tag 'message_icon', class: 'sidemenu-icon'%>
```
Comment (micaelalorenzo) [inline app/views/users/_sidemenu.html.erb]: Same here

File: app/views/users/_sidemenu.html.erb [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+        <h2 class='sidemenu-title'>Messages</h2>
+      </div>
+
+      <div class="sidemenu-item">
+        <%= image_tag 'profile_icon', class: 'sidemenu-icon'%>
```
Comment (micaelalorenzo) [inline app/views/users/_sidemenu.html.erb]: Same here

File: app/views/users/_sidemenu.html.erb [hunk 20]
```diff
@@ -0,0 +1,51 @@
...
+        <h2 class='sidemenu-title'>Profile</h2>
+      </div>
+
+      <div class="sidemenu-item">
+        <%= image_tag 'more_icon', class: 'sidemenu-icon'%>
```
Comment (micaelalorenzo) [inline app/views/users/_sidemenu.html.erb]: Same here

File: app/views/users/_sidemenu.html.erb [hunk 21]
```diff
@@ -0,0 +1,51 @@
...
+        <h2 class='sidemenu-title'>More</h2>
+      </div>
+    </div>
+
+    <div class="flex flex-col gap-3 items-start py-4">
+      <%= button_to 'Tweet', '#', class: 'btn btn-lg btn-primary px-24 xl:block hidden' %>
+      <%= image_tag 'add_tweet_icon', class: 'block xl:hidden w-12 h-12'%>
```
Comment (micaelalorenzo) [inline app/views/users/_sidemenu.html.erb]: Same here

File: app/views/users/edit.html.erb [hunk 22]
```diff
@@ -0,0 +1,46 @@
...
+        }%>
+        </div>
+
+        <h2 class="heading-3 m-6">Edit Profile</h2> 
+      </div>
+
+      <div class="h-[12.5rem] bg-dark-300">
+      </div>
+
+      <%= image_tag 'profile_picture', class: 'profile-picture ml-3.5'%>
```
Comment (micaelalorenzo) [inline app/views/users/edit.html.erb]: Same here

File: app/views/users/edit.html.erb [hunk 23]
```diff
@@ -0,0 +1,46 @@
...
+  <div class="w-screen h-screen absolute flex bg-dark-200/90">
+    <div class='w-[37.5rem] h-[55.5rem] m-auto bg-white rounded-3xl shadow-lg'>
+
+      <div class="flex w-full h-[4.188rem] items-center">
+        <div class="flex-col flex-none m-6">
+          <%= link_to image_tag('close_icon'), '#', data: {
+            turbo_frame: "_top",
+            controler: "modals",
+            action: "modals#close"
+        }%>
```
Comment (micaelalorenzo) [inline app/views/users/edit.html.erb]: Same here

File: app/views/user/edit.html.erb [hunk 24]
```diff
@@ -0,0 +1,47 @@
...
+      <%= image_tag 'profile_picture', class: 'profile-picture ml-3.5' %>
+
+      <%= form_with(model: @user, html: { id: dom_id(@user), class: 'max-w-[30.2rem] w-full h-full mx-14 mt-9 flex flex-col gap-y-5' }, :url =>{ :action => "update" }, method:'put' ) do |f| %>
+
+        <div class="w-full">
+          <%= f.text_field :name, autofocus: true, autocomplete: "name", placeholder: 'Name', class: 'input' %>
+        </div>
+
+        <div class="w-full">
+          <%= f.text_area :bio, autocomplete: "bio", placeholder: 'Bio', row: 2 ,class: 'input  h-[7.5rem]' %>
```
Comment (martin8767) [inline app/views/user/edit.html.erb]:
```html.erb
<%= f.text_area :bio, autocomplete: "bio", placeholder: 'Bio', row: 2, class: 'input h-[7.5rem]' %>
```

**Review by martin8767 — COMMENTED** [review]
GJ, just a small comment

**Pull request merged**